tests/components/homekit/test_type_cameras.py

```diff
@@ -0,0 +1,352 @@
+        2,
+        {CONF_STREAM_SOURCE: "/dev/null", CONF_SUPPORT_AUDIO: True},
+    )
+    not_camera_acc = Switch(hass, run_driver, "Switch", entity_id, 4, {},)
+    bridge = HomeBridge("hass", run_driver, "Test Bridge")
+    bridge.add_accessory(acc)
+    bridge.add_accessory(not_camera_acc)
+
+    await acc.run_handler()
+
+    assert acc.aid == 2
+    assert acc.category == 17  # Camera
+
+    stream_service = acc.get_service("CameraRTPStreamManagement")
+    endpoints_config_char = stream_service.get_characteristic("SetupEndpoints")
+    assert endpoints_config_char.setter_callback
+    stream_config_char = stream_service.get_characteristic(
+        "SelectedRTPStreamConfiguration"
+    )
+    assert stream_config_char.setter_callback
+    acc.set_endpoints(
+        "ARAzA9UDF8xGmrZykkNqcaL2AgEAAxoBAQACDTE5Mi4xNjguMjA4LjUDAi7IBAKkxwQlAQEAAhDN0+Y0tZ4jzoO0ske9UsjpAw6D76oVXnoi7DbawIG4CwUlAQEAAhCyGcROB8P7vFRDzNF2xrK1Aw6NdcLugju9yCfkWVSaVAYEDoAsAAcEpxV8AA=="
+    )
+
+    with patch(
+        "homeassistant.components.demo.camera.DemoCamera.stream_source",
+        return_value=None,
+    ), patch(
+        "homeassistant.components.homekit.type_cameras.HAFFmpeg",
+        return_value=_get_working_mock_ffmpeg(),
+    ):
+        acc.set_selected_stream_configuration(
+            "ARUCAQEBEDMD1QMXzEaatnKSQ2pxovYCNAEBAAIJAQECAgECAwEAAwsBAgAFAgLQAgMBHgQXAQFjAgQ768/RAwIrAQQEAAAAPwUCYgUDLAEBAwIMAQEBAgEAAwECBAEUAxYBAW4CBCzq28sDAhgABAQAAKBABgENBAEA"
+        )
+        await hass.async_block_till_done()
+
+    session_info = {
+        "id": "mock",
+        "v_srtp_key": "key",
+        "a_srtp_key": "key",
+        "v_port": "0",
+        "a_port": "0",
+        "address": "0.0.0.0",
+    }
+    acc.sessions[UUID("3303d503-17cc-469a-b672-92436a71a2f6")] = session_info
+
+    with patch(
+        "homeassistant.components.demo.camera.DemoCamera.stream_source",
+        return_value="rtsp://example.local",
+    ), patch(
+        "homeassistant.components.homekit.type_cameras.HAFFmpeg",
+        return_value=_get_working_mock_ffmpeg(),
+    ):
+        acc.set_selected_stream_configuration(
+            "ARUCAQEBEDMD1QMXzEaatnKSQ2pxovYCNAEBAAIJAQECAgECAwEAAwsBAgAFAgLQAgMBHgQXAQFjAgQ768/RAwIrAQQEAAAAPwUCYgUDLAEBAwIMAQEBAgEAAwECBAEUAxYBAW4CBCzq28sDAhgABAQAAKBABgENBAEA"
+        )
+        await acc.stop_stream(session_info)
+        # Calling a second time should not throw
+        await acc.stop_stream(session_info)
+        await hass.async_block_till_done()
+
+    assert await hass.async_add_executor_job(acc.get_snapshot, 1024)
+
+    # Verify the bridge only forwards get_snapshot for
+    # cameras and valid accessory ids
+    assert await hass.async_add_executor_job(bridge.get_snapshot, {"aid": 2})
+    with pytest.raises(ValueError):
+        assert await hass.async_add_executor_job(bridge.get_snapshot, {"aid": 3})
+    with pytest.raises(ValueError):
+        assert await hass.async_add_executor_job(bridge.get_snapshot, {"aid": 4})
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)